src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -2,54 +2,158 @@
+
+  private DebugEntry<Boolean> shooterReadyEntry;
+
+  private ShuffleboardTab shooterTab = Shuffleboard.getTab("Shooter fire");
+  private GenericEntry targetRPM = shooterTab.add("Target RPM", 0).getEntry();
+
+  private SpeakerConfig targetSpeeds;
+
+  private FlywheelSim shooterLeftSim;
+  private FlywheelSim shooterRightSim;
 
   public ShooterSubsystem() {
-    shooterTopMotor =
-        new CANSparkMax(Constants.ShooterConstants.SHOOTER_MOTOR_TOP_ID, MotorType.kBrushless);
-    shooterBottomMotor =
-        new CANSparkMax(Constants.ShooterConstants.SHOOTER_MOTOR_BOTTOM_ID, MotorType.kBrushless);
-
-    shooterTopMotor.restoreFactoryDefaults();
-    shooterTopMotor.setSmartCurrentLimit(40);
-    shooterBottomMotor.restoreFactoryDefaults();
-    shooterBottomMotor.setSmartCurrentLimit(40);
-
-    shooterTopMotor.getPIDController().setP(Constants.ShooterConstants.SHOOTER_P);
-    shooterTopMotor.getPIDController().setI(Constants.ShooterConstants.SHOOTER_I);
-    shooterTopMotor.getPIDController().setD(Constants.ShooterConstants.SHOOTER_D);
-    shooterTopMotor.getPIDController().setFF(Constants.ShooterConstants.SHOOTER_FF);
-    shooterBottomMotor.getPIDController().setP(Constants.ShooterConstants.SHOOTER_P);
-    shooterBottomMotor.getPIDController().setI(Constants.ShooterConstants.SHOOTER_I);
-    shooterBottomMotor.getPIDController().setD(Constants.ShooterConstants.SHOOTER_D);
-    shooterBottomMotor.getPIDController().setFF(Constants.ShooterConstants.SHOOTER_FF);
+    shooterLeftMotor =
+        new CANSparkMaxSim(Constants.ShooterConstants.SHOOTER_MOTOR_LEFT_ID, MotorType.kBrushless);
+    shooterRightMotor =
+        new CANSparkMaxSim(Constants.ShooterConstants.SHOOTER_MOTOR_RIGHT_ID, MotorType.kBrushless);
+
+    shooterLeftMotor.restoreFactoryDefaults();
+    shooterLeftMotor.setSmartCurrentLimit(40);
+    shooterRightMotor.restoreFactoryDefaults();
+    shooterRightMotor.setSmartCurrentLimit(40);
+
+    shooterLeftMotor.setInverted(true);
+
+    shooterLeftMotor.getPIDController().setP(Constants.ShooterConstants.SHOOTER_LEFT_P);
+    shooterLeftMotor.getPIDController().setI(Constants.ShooterConstants.SHOOTER_LEFT_I);
+    shooterLeftMotor.getPIDController().setD(Constants.ShooterConstants.SHOOTER_LEFT_D);
+    shooterLeftMotor.getPIDController().setFF(Constants.ShooterConstants.SHOOTER_LEFT_FF);
+    shooterRightMotor.getPIDController().setP(Constants.ShooterConstants.SHOOTER_RIGHT_P);
+    shooterRightMotor.getPIDController().setI(Constants.ShooterConstants.SHOOTER_RIGHT_I);
+    shooterRightMotor.getPIDController().setD(Constants.ShooterConstants.SHOOTER_RIGHT_D);
+    shooterRightMotor.getPIDController().setFF(Constants.ShooterConstants.SHOOTER_RIGHT_FF);
+
+    if (DriverStation.isTest()) {
+      ShooterTab.add("Shooter Left Motor PID", shooterLeftMotor.getPIDController());
+      ShooterTab.add("Shooter Right Motor PID", shooterRightMotor.getPIDController());
+    }
+
```

Comment:
We probably want these outside of test mode as well

Reply:
Resolved in 8c82a38